pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -365,11 +366,11 @@ func (m *managerImpl) processShutdownEvent() error {
 			go func(pod *v1.Pod, group podShutdownGroup) {
 				defer wg.Done()
 
-				gracePeriodOverride := group.ShutdownGracePeriodSeconds
+				gracePeriodOverride := int64(v1.DefaultTerminationGracePeriodSeconds)
 
```

Comment:
This seems to ignore the configured graceful shutdown period, which I think is a broken change that should require discussion and a KEP to design what you're going to do.

/hold

Reply:
``` go
// Set a default value for the TerminationGracePeriodSeconds
var terminationGracePeriodSeconds = int64(v1.DefaultTerminationGracePeriodSeconds)
if pod.Spec.TerminationGracePeriodSeconds != nil {
  terminationGracePeriodSeconds = *pod.Spec.TerminationGracePeriodSeconds
}

gracePeriodOverride := group.ShutdownGracePeriodSeconds

gracePeriodOverride = integer.Int64Min(gracePeriodOverride, gracePeriodOverride)
```

The logic might be clearer with this change

Reply:
Thanks. I like the first option the best. I will bring up this PR in sig node today.

Reply:
@wzshiming The graceful shutdown period should be the worst case; not the default case.

ie: Setting graceful shutdown period to 1h, should still apply the 30s default grace period if the pod's terminationGracePeriodSeconds is not set.

Reply:
@wzshiming We discussed in sig-node and came to a consensus this is the right approach.

/hold cancel